Makefile.gcc4

```diff
@@ -0,0 +1,33 @@
+# Makefile for ancient gcc4 (tested on Ubuntu 7.10 with 4.1.2-16ubuntu2)
+# Compiles Kaitai Struct C++ runtime and tests
+
+BUILD_DIR := build
+
+SOURCES := \
+	kaitai/kaitaistream.cpp \
+	tests/unittest.cpp
+
+OBJS := \
+	$(BUILD_DIR)/kaitaistream.o \
+	$(BUILD_DIR)/unittest.o
+
+DEFINES := -DKS_STR_ENCODING_ICONV -DGTEST_NANO
+
+LFLAGS :=
+LIBS :=
+
+INCLUDES := -I.
+
+all: $(BUILD_DIR) $(BUILD_DIR)/unittest
+
+$(BUILD_DIR):
+	mkdir -p $(BUILD_DIR)
+
+$(BUILD_DIR)/unittest: $(OBJS)
+	$(CXX) $(CXXFLAGS) -o build/unittest $(OBJS) $(LFLAGS) $(LIBS)
+
+$(BUILD_DIR)/kaitaistream.o: kaitai/kaitaistream.cpp
+	$(CXX) $(CXXFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@
+
+$(BUILD_DIR)/unittest.o: tests/unittest.cpp tests/gtest-nano.h
+	$(CXX) $(CXXFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@
```

Comment:
Is there a reason you specify `tests/gtest-nano.h` as a dependency of `unittest.o`, but not `kaitai/kaitaistream.h` which both object files depend on?

```suggestion
$(BUILD_DIR)/kaitaistream.o: kaitai/kaitaistream.cpp kaitai/kaitaistream.h
	$(CXX) $(CXXFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@

$(BUILD_DIR)/unittest.o: tests/unittest.cpp tests/gtest-nano.h kaitai/kaitaistream.h
	$(CXX) $(CXXFLAGS) $(INCLUDES) $(DEFINES) -c $< -o $@
```

Technically, if the `kaitai/kaitaistream.h` file is modified, both object files have to be recompiled.

Reply:
Nope, just forgot about that. Thanks for spotting!